plugin.video.reddit_viewer/default.py

```diff
@@ -1270,480 +669,291 @@ def listVideos(url, name, subreddit):
+        xbmc.executebuiltin('Container.SetViewMode(%s)' %viewID )
+    else:
+        if forceViewMode:
+            xbmc.executebuiltin('Container.SetViewMode('+viewMode+')')
 
+    xbmcplugin.endOfDirectory(handle=pluginhandle,
+                              succeeded=True,
+                              updateListing=False,   #setting this to True causes the ".." entry to quit the plugin
+                              cacheToDisc=True)
 
-def addLink(title, title_line2, iconimage, description, credate, reddit_says_is_video, site, subreddit, media_url, over_18, posted_by="", num_comments=0,post_index=1,post_total=1 ):
+def addLink(title, title_line2, iconimage, previewimage,preview_w,preview_h,domain,description, credate, reddit_says_is_video, commentsUrl, subreddit, media_url, over_18, posted_by="", num_comments=0,post_index=1,post_total=1,many_subreddit=False ):
+    #from resources.lib.utils import ret_info_type_icon, assemble_reddit_filter_string,build_script
+    from resources.lib.domains import parse_reddit_link, build_DirectoryItem_url_based_on_media_type
 
     videoID=""
     post_title=title
+    il_description=""
     n=""  #will hold red nsfw asterisk string
     h=""  #will hold bold hoster:  string
-    ok = False    
-    #DirectoryItem_url=""
-    hoster, DirectoryItem_url, videoID, mode_type, thumb_url, poster_url, isFolder,setInfo_type, IsPlayable=make_addon_url_from(media_url,reddit_says_is_video)
-    name_param='' #extra parameter sent to this addon to put a 'title' on slideshows
+    t_Album = translation(30073) if translation(30073) else "Album"
+    t_IMG =  translation(30074) if translation(30074) else "IMG"
 
-    if iconimage in {"","nsfw", "default"}:
-        #log( title+ ' iconimage blank['+iconimage+']['+thumb_url+ ']media_url:'+media_url ) 
+    ok = False
+
+    isFolder=True
+    thumb_url=''
+
+    h="[B]" + domain + "[/B]: "
+    if over_18:
+        mpaa="R"
+        n = "[COLOR red]*[/COLOR] "
+        #description = "[B]" + hoster + "[/B]:[COLOR red][NSFW][/COLOR] "+title+"\n" + description
+        il_description = "[COLOR red][NSFW][/COLOR] "+ h+title+"[CR]" + "[COLOR grey]" + description + "[/COLOR]"
+    else:
+        mpaa=""
+        n=""
+        il_description = h+title+"[CR]" + "[COLOR grey]" + description + "[/COLOR]"
+
+    if TitleAddtlInfo:     #put the additional info on the description if setting set to single line titles
+        log( repr(title_line2 ))
+        post_title=n+title+"[CR]"+title_line2
+    else:
+        post_title=n+title
+        il_description=title_line2+"[CR]"+il_description
+
+    il={"title": post_title, "plot": il_description, "plotoutline": il_description, "Aired": credate, "mpaa": mpaa, "Genre": "r/"+subreddit, "studio": domain, "director": posted_by }   #, "duration": 1271}   (duration uses seconds for titan skin
+
+    liz=xbmcgui.ListItem(label=post_title,
+                          label2="",
+                          iconImage="",
```

Comment:
iconImage and thumbnailImage are both deprecated use liz.setArt instead with a dictionary of image resources